### DISCUSSION TITLE
How connections are created and maintained with rueidis?

### DISCUSSION THREAD

**Discussion by sathishdv** [redis/rueidis#357] [Q&A] [answered]
Hi, I am using rueidis client to interact with AWS memory db, and when the new connection is established and work is performed then after that the connection remains open indefinitely, and I see bunch of open connections which never got to closed even when there's no ops performed. Is this something expected? I wanted to know if this how it suppose to be or am I doing something wrong. I am worried if this causes any memory/connection leaks. Any pointers/help greatly appreciated :)

**Comment by rueian** [comment, marked as answer]
Hi @sathishdv,

You are not doing anything wrong. Connections are reused and are not closed unless they encounter network errors.

There are two options you can adjust number of connections to each redis node:
1. PipelineMultiplex: this option can adjust how many  connections used for pipelining.
2. BlockingPoolSize: this option can adjust maximum number of connections used by dedicated or blocking commands.

**Reply by cjo2** [reply]
Hi, thanks @rueian. It wasn't clear to me that BlockingPoolSize also affected the Dedicated connections, but in my testing not too long ago, it looks like that parameter affected the connection count. This was super helpful.

**Reply by sathishdv** [reply]
Thanks @rueian this helps!
